stanford-earth/SE3_BLT
# EARTH-256: Add behat config with updated project.yml

# NOT READY

# Summary
- Changes to the `blt/project.yml` file should help BLT generate a local.yml file that includes additional paths, like `/modules` and `/profiles`, to look for behat tests.
- It also allows us to output the results as an html page, in addition to the `pretty` progress report.
- We discussed replacing `tests:behat` or creating a new, `sws:behat`. I like creating an `sws:` prefix for BLT commands we override.  That way, if we don't have to wonder, did we end up overriding this?  It's incredibly clear.
- This PR also includes a working .travis.yml file, up to site install due to a lack of ssh keys for pulling from production.

# Urgency
- I have included in this pull request the rc version of caravan, so there is no immediate urgency to close.

# Steps to Test
- Setup requires a bit of work.  Checkout this branch and `run composer update`.
- Place a feature file, somewhere like: `docroot/modules/stanford/stanford_news/tests/stanford_news.feature` from https://raw.githubusercontent.com/SU-SWS/linky_clicky/5.x/sites/uat/features/high-value/stanford_news.feature.
- Run `blt sws:behat` and it should find the stanford_news.feature file.
- You should be able to see output from the tests as they run and afterwards find the full results at http://[local_url]/reports/behat/index.html.

# Affected versions or modules
- https://github.com/SU-SWS/drupal-caravan/releases/tag/1.0.0-rc

# Associated Issues and/or People
- JIRA ticket: https://stanfordits.atlassian.net/browse/EARTH-256
